homeassistant/components/plugwise/climate.py

```diff
@@ -0,0 +1,310 @@
+        return _attributes
+
+    def update(self):
+        """Update the data from the thermostat."""
+        _LOGGER.debug("Update called")
+        domain_objects = self._api.get_domain_objects()
+        self._current_temperature = self._api.get_temperature(domain_objects)
+        self._temperature = self._api.get_target_temperature(domain_objects)
+        self._preset_mode = self._api.get_current_preset(domain_objects)
+        self._outdoor_temperature = self._api.get_outdoor_temperature(
+            domain_objects)
+        self._heating_action = _heating_action(self)
+        self._available_schemas = _available_schemas(self)
+        if self._active_schema != _active_schema(self):
+            self._previous_schema = self._active_schema
+            self._active_schema = _active_schema(self)
+        self._hvac_mode = _hvac_mode(self)
+        self._preset_modes = []
+        for preset in [*self._api.get_presets(domain_objects)]:
+            self._preset_modes.append(preset)
+
+    @property
+    def hvac_mode(self):
+        """Return current active hvac state."""
+        return _hvac_mode(self)
+
+    @property
+    def heating_action(self):
+        """Return current active heating state."""
+        return _heating_action(self)
+
+    @property
+    def preset_mode(self):
+        """Return the active preset mode."""
+        return self._preset_mode
+
+    @property
+    def preset_modes(self):
+        """Return the available preset modes list."""
+        return self._preset_modes
+
+    @property
+    def hvac_modes(self):
+        """Return the available hvac  modes list."""
+        return self._hvac_modes
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def min_temp(self):
+        """Return the minimal temperature possible to set."""
+        return self._min_temp
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature possible to set."""
+        return self._max_temp
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature."""
+        return self._temperature
+
+    @property
+    def outdoor_temperature(self):
+        """Return the outdoor temperature."""
+        return self._outdoor_temperature
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measured temperature."""
+        return TEMP_CELSIUS
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        _LOGGER.debug("Adjusting temperature")
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if (temperature is not None and self._min_temp < temperature
+                < self._max_temp):
+            self._temperature = temperature
+            domain_objects = self._api.get_domain_objects()
+            self._api.set_temperature(domain_objects, temperature)
+            self.schedule_update_ha_state()
```

Comment:
This is implicit and not needed when should_poll is true

Reply:
Something learned, so that part is actually for async, thanks!